src/App.js

```diff
@@ -3,60 +3,108 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
-const App = () => {
-  const [todoList, setTodoList] = React.useState([]);
-  const [isLoading, setIsLoading] = React.useState(true);
 
+const AIRTABLE_API_URL = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
 
-  React.useEffect(() => {
-    new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve({
-          data: {
-            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []
-          }
-        })
-      }, 2000)
-    })
-      .then((result) => {
-        setTodoList(result.data.todoList);
-        setIsLoading(false);
-      });
+function App() {
+
+  const [todoList, setTodoList] = React.useState([])
+  const [isloading, setIsLoading] = React.useState(true)
+
+  const fetchData = React.useCallback(async () => {
+
+    const options = {
+      method: 'GET',
+      headers: { Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    try {
+      const response = await fetch(AIRTABLE_API_URL, options)
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.title, id: todo.id, }
+      })
+      setTodoList(todos)
+      setIsLoading(false)
+    } catch (error) {
+      console.log(error.message)
+    }
   }, [])
 
-  React.useEffect(() => {
-    if (!isLoading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+  const addTodo = async ({ title }) => {
```

Comment:
Great job implementing this functionality! Something to consider: this function is doing a lot. It's posting to the API and it's also updating your state. Consider separating those concerns and streamlining the functionality of this function.